### PR TITLE
feat: define a generic no-op table cell parser

### PR DESCRIPTION
**Pull request by arjunlalb**
## Description
Generic object table cell parser

### Testing
None

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
A generic table cell parser for no-op object parser usecases.

### CODE REVIEW

File: projects/components/src/table/cells/data-parsers/table-cell-object-parser.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { TableCellParser } from '../table-cell-parser';
+import { TableCellParserBase } from '../table-cell-parser-base';
+import { CoreTableCellParserType } from '../types/core-table-cell-parser-type';
+
+@TableCellParser({
+  type: CoreTableCellParserType.Object
+})
+export class TableCellObjectParser extends TableCellParserBase<ObjectCellData, ObjectCellData, string | undefined> {
```
Comment (aaron-steinfeld) [inline projects/components/src/table/cells/data-parsers/table-cell-object-parser.ts:8]: Could we make this even more generic? For example, why does it have to even be an object?

The parseValue function could remain as is, and the parseFilterValue function could just have a bit of logic like
```typescript
if (typeof cellData === 'object' && cellData !== null && 'filterValue' in cellData) {
  return cellData.filterValue;
}
return String(cellData);
```

File: projects/components/src/table/cells/data-parsers/table-cell-object-parser.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    return cellData;
+  }
+
+  public parseFilterValue(cellData: ObjectCellData): string | undefined {
+    return this.parseValue(cellData).filterValue;
+  }
+}
+
+// Since object is very generic, using an explicit key fr default filter value
+interface ObjectCellData extends Object {
```
Comment (aaron-steinfeld) [inline projects/components/src/table/cells/data-parsers/table-cell-object-parser.ts:19]: `Object` should only be used as a constructor, instead use lowercased `object` when referencing a type (similar to `String` and `string`)

File: projects/components/src/table/cells/data-parsers/table-cell-object-parser.ts (hunk 1 above)
Comment (arjunlalb) [reply projects/components/src/table/cells/data-parsers/table-cell-object-parser.ts:8]: Made everything unknown.

File: projects/components/src/table/cells/types/core-table-cell-parser-type.ts [hunk 3]
```diff
@@ -4,5 +4,6 @@ export const enum CoreTableCellParserType {
   Number = 'number',
   String = 'string',
   Timestamp = 'timestamp',
-  Icon = 'icon'
+  Icon = 'icon',
+  Object = 'object'
```
Comment (aaron-steinfeld) [inline projects/components/src/table/cells/types/core-table-cell-parser-type.ts:8]: This value would update too. Do you prefer the name generic over NoOp ? I feel like Noop describes what it's doing better

File: projects/components/src/table/cells/types/core-table-cell-parser-type.ts [hunk 4]
```diff
@@ -5,5 +5,5 @@ export const enum CoreTableCellParserType {
   String = 'string',
   Timestamp = 'timestamp',
   Icon = 'icon',
-  Object = 'object'
+  NoOp = 'noOp'
```
Comment (aaron-steinfeld) [inline projects/components/src/table/cells/types/core-table-cell-parser-type.ts:8]: Nittiest nit:
```suggestion
  NoOp = 'noop'
```

**Review by aaron-steinfeld — APPROVED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**